pkg/controllers/inventory_controller.go

```diff
@@ -244,108 +245,64 @@ func (r *InventoryReconciler) SetupWithManager(mgr ctrl.Manager) error {
-					"inventory": i.Name,
-				},
-			},
-
-			Spec: rufio.BMCJobSpec{
-				BaseboardManagementRef: rufio.BaseboardManagementRef{
-					Name:      i.Name,
-					Namespace: i.Namespace,
-				},
-				Tasks: []rufio.Task{
-					{
-						PowerAction: &off,
-					},
-					{
-						OneTimeBootDeviceAction: &rufio.OneTimeBootDeviceAction{
-							Devices: []rufio.BootDevice{
-								rufio.PXE,
-							},
-							EFIBoot: false,
-						},
-					},
-					{
-						PowerAction: &on,
-					},
-				},
-			},
-		}
-		err := controllerutil.SetOwnerReference(i, job, r.Scheme)
-		if err != nil {
-			return err
-		}
-		err = r.Create(ctx, job)
-		if err != nil {
-			return err
-		}
-
-		i.Status.Conditions = util.CreateOrUpdateCondition(i.Status.Conditions, seederv1alpha1.BMCJobSubmitted, "BMCJob submitted")
-
-		return r.Status().Update(ctx, i)
+		i.Spec.PowerActionRequested = seederv1alpha1.NodePowerActionReboot
+		return r.Update(ctx, i)
 	}
 
 	return nil
 }
 
 // reconcileBMCJob will update the BMCJob conditions to reflect current state of the job for specific inventory
-func (r *InventoryReconciler) reconcileBMCJob(ctx context.Context, i *seederv1alpha1.Inventory) error {
-
+func (r *InventoryReconciler) reconcileBMCJob(ctx context.Context, iObj *seederv1alpha1.Inventory) error {
+	i := iObj.DeepCopy()
 	if util.ConditionExists(i.Status.Conditions, seederv1alpha1.BMCJobSubmitted) {
-		j := &rufio.BMCJob{}
-		var jobFound bool
-		err := r.Get(ctx, types.NamespacedName{Namespace: i.Namespace, Name: fmt.Sprintf("%s-reboot", i.Name)}, j)
+		j := &rufio.Job{}
+		var completed bool
+		err := r.Get(ctx, types.NamespacedName{Namespace: i.Namespace, Name: i.Status.PowerAction.LastJobName}, j)
 		if err != nil {
-			if !apierrors.IsNotFound(err) {
-				return err
-			}
-		} else {
-			jobFound = true
+			return err
 		}
 
-		if jobFound {
-			if j.HasCondition(rufio.JobCompleted, rufio.ConditionTrue) {
-				i.Status.Conditions = util.CreateOrUpdateCondition(i.Status.Conditions, seederv1alpha1.BMCJobComplete, "")
-			}
+		if j.HasCondition(rufio.JobCompleted, rufio.ConditionTrue) {
+			i.Status.Conditions = util.CreateOrUpdateCondition(i.Status.Conditions, seederv1alpha1.BMCJobComplete, "")
+			i.Status.PowerAction.LastActionStatus = seederv1alpha1.NodeJobComplete
+			completed = true
+		}
 
-			if j.HasCondition(rufio.JobFailed, rufio.ConditionTrue) {
-				var message string
-				for _, c := range j.Status.Conditions {
-					if c.Type == rufio.JobFailed && c.Status == rufio.ConditionTrue {
-						message = c.Message
-					}
+		if j.HasCondition(rufio.JobFailed, rufio.ConditionTrue) {
+			var message string
+			for _, c := range j.Status.Conditions {
+				if c.Type == rufio.JobFailed && c.Status == rufio.ConditionTrue {
+					message = c.Message
 				}
-				i.Status.Conditions = util.CreateOrUpdateCondition(i.Status.Conditions, seederv1alpha1.BMCJobError, message)
+				i.Status.PowerAction.LastActionStatus = seederv1alpha1.NodeJobFailed
```

Comment:
It seems unnecessary to put this line inside the for loop.

Reply:
great spot. This is not needed inside the inner for loop. fixed.